sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs

```diff
@@ -111,28 +128,29 @@ private static void Initialize()
                     initialized = true;
                 }
             }
-            GlobalLog("Before initialize");
+            Log("Before initialize");
             var watch = Stopwatch.StartNew();
             if (cbdeBinaryPath != null)
             {
                 RegisterMlirAndCbdeInOneStep(context);
             }
             watch.Stop();
-            GlobalLog($"After initialize ({watch.ElapsedMilliseconds} ms)");
+            Log($"After initialize ({watch.ElapsedMilliseconds} ms)");
         }
 
         private void RegisterMlirAndCbdeInOneStep(SonarAnalysisContext context)
         {
             context.RegisterCompilationAction(
                 c =>
                 {
-                    if(!shouldRunInContext(c))
+                    emitLog = Environment.GetEnvironmentVariables().Contains("SONAR_DOTNET_LOG_CBDE");
```

Comment:
small suggestion: put _internal_ in the name - `SONAR_DOTNET_INTERNAL_LOG_CBDE` , to make it clear it's for our internal usage